### PR TITLE
User profile fields

### PR DESCRIPTION
**Pull request by jonathanwgoodwin**
These don't appear to be documented but do appear to exist. AFAICT its a `Map<String, ProfileField>`.

For  us for example we've got the actual username of the user in here, as well as other visible / non-visible custom profile data. 

Example redacted data:
```json
{
  "fields": {
    "X000000000": {
       "value": "",
       "alt": ""
     },
  }
}
```

### CODE REVIEW

File: slack-base/src/main/java/com/hubspot/slack/client/models/users/UserProfileIF.java [hunk 1]
```diff
@@ -28,6 +29,9 @@
   Optional<String> getPhone();
   Optional<String> getSkype();
 
+  // Extra custom fields set by your workspace admin
+  Optional<Map<String, ProfileField>> getFields();
```
Comment (darcatron) [inline slack-base/src/main/java/com/hubspot/slack/client/models/users/UserProfileIF.java]: Would be awesome if we could re-use [this deserialization test](https://github.com/HubSpot/slack-client/blob/master/slack-base/src/test/java/com/hubspot/slack/client/models/users/SlackUserDeserializationTest.java) to verify these fields are parsed

File: slack-base/src/main/java/com/hubspot/slack/client/models/users/ProfileFieldIF.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.hubspot.immutables.style.HubSpotStyle;
+
+@Immutable
+@HubSpotStyle
+public interface ProfileFieldIF {
+
+  String getValue();
+
+  String getAlt();
```
Comment (johnnyleitrim) [inline slack-base/src/main/java/com/hubspot/slack/client/models/users/ProfileFieldIF.java]: Sounds like the `alt` field is [optional](https://api.slack.com/methods/users.profile.set):
> The `value` of a field is what should be displayed, unless the `alt` key is also present, in which case that is displayed instead.

**Comment by szabowexler** [comment]
@jonathanwgoodwin what's the story here? Is this tested/ok to merge/still necessary? I'd like to either get it merged in or closed.

**Comment by jonathanwgoodwin** [comment]
Turns out that this isn’t sent in the data for the list users call, only when you get an individual user account. That made the thing I wanted to use this for too slow so I’ll close this for now.

**Comment by jonathanwgoodwin** [comment]
I'm back  🤣, the original use-case for this field didn't work out, but I've got a new place where I need to access these profile fields.

**Pull request merged**